src/lib/request-with-rate-limit.ts

```diff
@@ -0,0 +1,37 @@
+import * as needle from 'needle';
+import Bottleneck from 'bottleneck';
+import { OutgoingHttpHeaders } from 'http2';
+
+export async function requestWithRateLimitRetries(
+    verb: needle.NeedleHttpVerbs,
+    url: string,
+    headers: OutgoingHttpHeaders,
+    limiter?: Bottleneck,
+  ): Promise<any> {
+    let data;
+    const maxRetries = 7;
+    let attempt = 0;
+    while (attempt < maxRetries) {
+        limiter ? limiter : limiter = new Bottleneck({ maxConcurrent: 1, minTime: 1000, });
+      data = await limiter.schedule(() => needle(verb, url, { headers: headers }));
+      if (data.statusCode === 200) {
```

Comment:
don't return here, we return at the end always, we need to stop the loop though

Break early if the codes do not retries

```
if ([404, 200].includes(data.statusCode){
    break;
}

```

Reply:
Done